src/expressions/array-expr.md

```diff
@@ -16,14 +16,21 @@ brackets. This produces an array containing each of these values in the
 order they are written.
 
 Alternatively there can be exactly two expressions inside the brackets,
-separated by a semi-colon. The expression after the `;` must have type
+separated by a semicolon. The expression after the `;` must have type
 `usize` and be a [constant expression],
 such as a [literal](../tokens.md#literals) or a [constant
 item](../items/constant-items.md). `[a; b]` creates an array containing `b`
-copies of the value of `a`. If the expression after the semi-colon has a value
+copies of the value of `a`. If the expression `b` after the semicolon has a value
 greater than 1 then this requires that the type of `a` is
 [`Copy`](../special-types-and-traits.md#copy), or `a` must be a path to a
-constant item.
+constant item. When `b` evaluates to 0, the expression `a` is evaluated once
```

Comment:
"the expression `a` is evaluated once" is *always* true for non-constant-items. So I think it is a bit strange to make `0` sound like a special case here.

I'd describe this (probably in a separate paragraph) as:
* for constant items, the expression is instantiated `b` times. (So if `b == 0`, it is not instantiated at all.)
* for other `a`, the expression is evaluated exactly once (and then the result is copied as needed).

Reply:
Oh, that makes sense and is much clearer to me now.  Thanks for your patience!